src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,248 @@
     private static final String ORDER_RECEIVED_GROUP_RETRY = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC_RETRY;
     private static final String ORDER_RECEIVED_GROUP_ERROR = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC_ERROR;
-    private static final int MAX_RETRY_ATTEMPTS = 3;
+    private static long ERROR_RECOVERY_OFFSET = 0l;
 
-    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupMain;
-    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupRetry;
     @Value("${spring.kafka.consumer.bootstrap-servers}")
-    private String brokerAddresses;
+    private String bootstrapServers;
+    @Value("${uk.gov.companieshouse.item-handler.error-consumer}")
+    private boolean errorConsumerEnabled;
     private final SerializerFactory serializerFactory;
+    private final OrdersKafkaProducer kafkaProducer;
+    private final KafkaListenerEndpointRegistry registry;
 
-    public OrdersKafkaConsumer(SerializerFactory serializerFactory) {
+    public OrdersKafkaConsumer(SerializerFactory serializerFactory,
+                               OrdersKafkaProducer kafkaProducer,
+                               KafkaListenerEndpointRegistry registry) {
         this.serializerFactory = serializerFactory;
+        this.kafkaProducer = kafkaProducer;
+        this.registry = registry;
     }
 
-    @Override
-    public void afterPropertiesSet() throws Exception {
-        LOGGER.debug("Initializing kafka consumer service " + this.toString());
-
-        ProducerConfig config = getProducerConfig();
-        CHKafkaProducer chKafkaProducer = new CHKafkaProducer(config);
-
-        ConsumerConfig consumerConfig = getConsumerConfig();
-
-        consumerConfig.setGroupName(ORDER_RECEIVED_GROUP);
-        consumerConfig.setTopics(Collections.singletonList(ORDER_RECEIVED_TOPIC));
-        chKafkaConsumerGroupMain
-                = new CHKafkaResilientConsumerGroup(consumerConfig, CHConsumerType.MAIN_CONSUMER,
-                        new KafkaConsumerFactory(), chKafkaProducer);
-
-        consumerConfig.setTopics(Collections.singletonList(ORDER_RECEIVED_TOPIC_RETRY));
-        consumerConfig.setGroupName(ORDER_RECEIVED_GROUP_RETRY);
-        chKafkaConsumerGroupRetry
-                = new CHKafkaResilientConsumerGroup(consumerConfig, CHConsumerType.RETRY_CONSUMER,
-                        new KafkaConsumerFactory(), chKafkaProducer);
-    }
-
-    @KafkaListener(topics = ORDER_RECEIVED_TOPIC, groupId = ORDER_RECEIVED_GROUP,
-            autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
-    public void processOrderReceived(String orderReceivedUri)
+    /**
+     * `order-received` topic listener/consumer. Calls a `retryable` method to process received message.
+     * If the `retryable` processor is unsuccessful with a `retryable` error, after maximum numbers of attempts allowed,
+     * the message is published to `order-received-retry` topic for failover processing.
```

Comment:
after maximum numbers of attempts allowed the message should be produced to the error topic. Are the topic names configurable? If so, they shouldn't be hard-coded in the JavaDocs.

Reply:
I have updated the failover processing and made changes to java doc and non-tryable failure logging.

Reply:
The above Javadoc does not seem to tally with the statement @jsimons1989 has made here?

Reply:
The JavaDoc describes the logical flow (implemented as described in the `Main Flow` section of `High Level Overview` [here](https://companieshouse.atlassian.net/wiki/spaces/DEV/pages/129564747/Kafka+resilience#Kafkaresilience-HighLevelOverview)). 
Also, the hard coded topic names have been replaced with generic suffixes `-retry` and `-error`.